Composer/packages/lib/indexers/src/qnaIndexer.ts

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+import LUParser from '@microsoft/bf-lu/lib/parser/lufile/luParser';
+import { FileInfo, QnAFile } from '@bfc/shared';
+import get from 'lodash/get';
+import { Diagnostic, Position, Range, DiagnosticSeverity } from '@bfc/shared';
+import { nanoid } from 'nanoid';
+
+import { getBaseName } from './utils/help';
+import { FileExtensions } from './utils/fileExtensions';
+
+function convertQnADiagnostic(d: any, source: string): Diagnostic {
+  const severityMap = {
+    ERROR: DiagnosticSeverity.Error,
+    WARN: DiagnosticSeverity.Warning,
+    INFORMATION: DiagnosticSeverity.Information,
+    HINT: DiagnosticSeverity.Hint,
+  };
+  const result = new Diagnostic(d.Message, source, severityMap[d.Severity]);
+
+  const start: Position = d.Range ? new Position(d.Range.Start.Line, d.Range.Start.Character) : new Position(0, 0);
+  const end: Position = d.Range ? new Position(d.Range.End.Line, d.Range.End.Character) : new Position(0, 0);
+  result.range = new Range(start, end);
+
+  return result;
+}
+
+function parse(content: string, id = '') {
+  const { Sections, Errors } = LUParser.parse(content);
+  const qnaSections: any[] = [];
+  Sections.forEach((section) => {
+    const {
+      Answer,
+      Body,
+      FilterPairs,
+      Id,
+      QAPairId,
+      Questions,
+      SectionType,
+      StartLine,
+      StopLine,
+      prompts,
+      promptsText,
+      source,
+    } = section;
```

Comment:
1. `const qnASections = Sections.map()`
2. It's not necessary to expand all values in `section`. Just override changed fields `{...section, range, Questions: ...}`